fix(license): validate license id and short name

maierthomas opened the pull request:
- add validate function to ThriftValidate.java
- verify license id / short name in addLicense and updateLicense

closes sw360/sw360portal#329

maierthomas commented:
`assertId` method in `SW360Asserts.java` should validate the correct document id?
How do I validate the doc id exactly?

mcjaeger commented:
please consider `1*(ALPHA / DIGIT / "-" / "." )`

from https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60

ID string is like "Apache-2.0" or "MIT" or "GPL-2.0-only"

mcjaeger commented:
(from discussion, Alex proposed) ... to use a new "assertLicenseId" not just assertId

maierthomas commented:
Should we include also `"+"` or is this character deprecated?
The validation of the value is only for a **new** license

maierthomas commented:
(from discussion) the character `"+"` is allowed

adrian-evo commented:
Tested and the license short name is validated fine.

alexbrdn commented on libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java:
```diff
@@ -244,6 +245,28 @@ public static void validateAttachment(AttachmentContent attachment) throws SW360
 
+    public static void validateLicenseIdMatch(License license) throws SW360Exception {
+        if (license.isSetId() && license.isSetShortname()) {
+            String message = "license short name must be equal with license id";
```
I believe "equal to" is the grammatically correct phrase

maierthomas replied:
corrected

alexbrdn commented on libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftValidate.java:
```diff
@@ -244,6 +245,28 @@ public static void validateAttachment(AttachmentContent attachment) throws SW360
+    }
+
+    public static void validateLicense(License license) throws SW360Exception {
+        final Pattern pattern = Pattern.compile("[A-Za-z0-9\\-.+]*");
```
please extract the regexp into a named constant

alexbrdn replied:
maybe even the whole pattern

maierthomas commented:
- renamed `validateLicense` to `validateNewLicense`
- removed if condition (isSet) for `license.shortname`
- removed assertId and pattern check for `license.id`
